crates/simulate/src/manager.rs

```diff
@@ -44,99 +53,115 @@ impl<U: UpdateData, W: World<WorldUpdateData = U>> SimManager<U, W> {
         }
     }
 
+    pub fn get_state_mut(&mut self) -> &mut State<U, W> {
+        &mut self.state
+    }
+
+    pub fn get_state(&self) -> &State<U, W> {
+        &self.state
+    }
+
     /// Run the time policy and agents to update the simulation environment.
     pub fn run_sim(&mut self) -> SimManagerResult<()> {
         // Initiate block time policy.
-        self.state
-            .update_time_env(self.time_policy.current_time_env());
+        self.state.update_time(self.time_policy.current_time_env());
 
         while self.time_policy.is_active() {
-            let (sender, receiver) = unbounded::<AgentSimUpdate<U>>();
+            let (evm_update_sender, evm_update_receiver) =
```

Comment:
Core loop is the same, but we now have separate channels for the evm and `World`. Also, we generate individual result receivers for each agent, so they can directly get the outputs of their state updates in the `resolve_step` function.